readthedocs/proxito/views/hosting.py

```diff
@@ -329,15 +329,24 @@ def _v0(self, project, version, build, filename, user):
                         {
                             # TODO: name this field "display_name"
                             "slug": translation.language,
-                            "url": f"/{translation.language}/",
+                            "url": resolver.resolve(
+                                project=project,
+                                version_slug=version.slug,
+                                language=translation.language,
+                                external=version.type == EXTERNAL,
+                            ),
                         }
                         for translation in project_translations
                     ],
                     "versions": [
                         {
                             # TODO: name this field "display_name"
                             "slug": version.slug,
-                            "url": f"/{project.language}/{version.slug}/",
+                            "url": resolver.resolve(
+                                project=project,
+                                version_slug=version.slug,
+                                external=version.type == EXTERNAL,
```

Comment:
This queryset has `.only("slug")`, since here it's accessing `type`, it'll probably do an extra query for each version.

https://github.com/readthedocs/readthedocs.org/blob/d93581e32e868985aab640930cbbbc127542f351/readthedocs/proxito/views/hosting.py#L226-L235

Reply:
@stsewd good catch! I'd like to add query check tests for these most important views, so we can catch stuff like this in dev. 💯

Reply:
I opened https://github.com/readthedocs/readthedocs.org/pull/10829 to solve this. Thanks for the review here.